iroh-willow/tests/basic.rs

```diff
@@ -0,0 +1,341 @@
+                SessionMode::ReconcileOnce,
+            );
+            let mut intent = alfie.sync_with_peer(betty_node_id, init).await.unwrap();
+
+            assert_eq!(
+                intent.next().await.unwrap(),
+                EventKind::CapabilityIntersection {
+                    namespace,
+                    area: Area::full(),
+                }
+            );
+
+            assert_eq!(
+                intent.next().await.unwrap(),
+                EventKind::InterestIntersection {
+                    namespace,
+                    area: Area::path(path.clone()).into()
+                }
+            );
+
+            assert_eq!(
+                intent.next().await.unwrap(),
+                EventKind::Reconciled {
+                    namespace,
+                    area: Area::path(path.clone()).into()
+                }
+            );
+
+            assert_eq!(intent.next().await.unwrap(), EventKind::ReconciledAll);
+
+            assert!(intent.next().await.is_none());
+        }
+    });
+
+    let task_bar_path = tokio::task::spawn({
+        let alfie = alfie.clone();
+        async move {
+            let path = Path::new(&[b"bar"]).unwrap();
+
+            let interests = Interests::builder().add_area(namespace, [Area::path(path.clone())]);
+            let init = SessionInit::new(interests, SessionMode::ReconcileOnce);
+
+            let mut intent = alfie.sync_with_peer(betty_node_id, init).await.unwrap();
+
+            assert_eq!(
+                intent.next().await.unwrap(),
+                EventKind::CapabilityIntersection {
+                    namespace,
+                    area: Area::full(),
+                }
+            );
+
+            assert_eq!(
+                intent.next().await.unwrap(),
+                EventKind::InterestIntersection {
+                    namespace,
+                    area: Area::path(path.clone()).into()
+                }
+            );
+
+            assert_eq!(
+                intent.next().await.unwrap(),
+                EventKind::Reconciled {
+                    namespace,
+                    area: Area::path(path.clone()).into()
+                }
+            );
+
+            assert_eq!(intent.next().await.unwrap(), EventKind::ReconciledAll);
+
+            assert!(intent.next().await.is_none());
+        }
+    });
+
+    task_foo_path.await.unwrap();
+    task_bar_path.await.unwrap();
+
+    [alfie, betty].map(Peer::shutdown).try_join().await?;
+
+    Ok(())
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn peer_manager_update_intent() -> Result<()> {
+    iroh_test::logging::setup_multithreaded();
+    let mut rng = create_rng("peer_manager_update_intent");
+
+    let [alfie, betty] = spawn_two(&mut rng).await?;
+    let (namespace, _alfie_user, betty_user) = setup_and_delegate(&alfie, &betty).await?;
+    let betty_node_id = betty.node_id();
+
+    insert(&betty, namespace, betty_user, &[b"foo"], "foo 1").await?;
```

Comment:
In that test case, if I change the path (both for inserted entry and interest area) to any path, longer than one component (e.g. `&[b"to", b"bar"]`), all session intents do work as expected, but actual entries are not synchronized for Alfie (her store is empty at the end of the test).

Reply:
Thanks for playing with this branch and spotting this!

It is very likely exactly as you make it out, a bug in the area-to-range conversion.

I will have a look in the next days. We're also planning to move towards using willow-rs for the data model types, so latest then this will be fixed for sure.